rust/tool/src/esp.rs

```diff
@@ -1,20 +1,62 @@
 use std::array::IntoIter;
 use std::path::{Path, PathBuf};
+use std::str::FromStr;
 
-use anyhow::{Context, Result};
+use anyhow::{Context, Result, Error, bail};
 
 use crate::generation::Generation;
 
+/// Supported system
+#[allow(dead_code)]
+#[non_exhaustive]
+pub enum System {
```

Comment:
I would prefer if this was `Arch` or `Architecture` instead of `System` because (a) `System` is a stupid name for it which nobody but Nix uses and (b) this enum really only contains the architecture part of the NixOS system double.